src/Reflection/ReflectionVariable.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace BetterReflection\Reflection;
+
+use phpDocumentor\Reflection\Type;
+use phpDocumentor\Reflection\Types;
+
+class ReflectionVariable
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var int
+     */
+    private $declaredAt;
+
+    /**
+     * @var $type
+     */
+    private $type;
+
+    /**
+     * @param string $name
+     * @param int $declaredAt
+     * @param string $type
+     * @return ReflectionType
+     */
+    public static function createFromName($name, ReflectionType $type = null, $declaredAt)
+    {
+        $reflectionType = new self();
+        $reflectionType->name = $name;
+        $reflectionType->declaredAt = $declaredAt;
+        $reflectionType->type = $type;
+        return $reflectionType;
+    }
+
+    public function getName() 
+    {
+        return $this->name;
+    }
+
+    public function getDeclaredAt() 
+    {
+        return $this->declaredAt;
+    }
+
+    /**
+     * Get a PhpDocumentor type object for this type
```

Comment:
What if the type mutated during execution? Two variables with same name, but different type and declaration line, then?